functions/go/create-setters/createsetters/create_setters.go

```diff
@@ -0,0 +1,275 @@
+package createsetters
+
+import (
+	"fmt"
+	"strings"
+
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/kio/kioutil"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+var _ kio.Filter = &CreateSetters{}
+
+// CreateSetters creates a comment for the resource fields which 
+// contains the same value as setter reference value 
+type CreateSetters struct {
+	// Setters holds the user provided values for simple map setters
+	Setters []Setter
+
+	// Setters holds the user provided values for array setters
+	ArraySetters []ArraySetter
+
+	// Results are the results of applying setter values
+	Results []*Result
+
+	// filePath file path of resource
+	filePath string
+}
+
+// Setter stores name and value of the map setter
+type Setter struct {
+	// Name is the name of the setter
+	Name string
+
+	// Value is the input value for setter
+	Value string
+}
+
+// ArraySetter stores name and values of the array setter
+type ArraySetter struct {
+	// Name is the name of the setter
+	Name string
+
+	// ValueSet is the set of the values for setter
+	ValueSet map[string]bool
+}
+
+// Result holds result of search and replace operation
+type Result struct {
+	// FilePath is the file path of the matching value
+	FilePath string
+
+	// FieldPath is field path of the matching value
+	FieldPath string
+
+	// Comment of the matching value
```

Comment:
This should hold field value and not comment. Add another variable `Comment` to this struct.

Reply:
Done

Reply:
It can just be the input string from user. Check here https://github.com/GoogleContainerTools/kpt-functions-catalog/blob/master/functions/go/apply-setters/applysetters/apply_setters.go#L170